lib/factory_bot/strategy_syntax_method_registrar.rb

```diff
@@ -1,37 +1,20 @@
 module FactoryBot
   # @api private
-  class StrategySyntaxMethodRegistrar
-    def initialize(strategy_name)
-      @strategy_name = strategy_name
+  module StrategySyntaxMethodRegistrar
+    def self.define_strategy_methods(strategy_name)
+      define_singular_strategy_method(strategy_name)
+      define_list_strategy_method(strategy_name)
+      define_pair_strategy_method(strategy_name)
     end
```

Comment:
Hum, I think I prefer the object oriented approach given how how `strategy_name` is passed around. I'd consider defining the `define_strategy_methods` class method as `new(strategy_name).define_strategy_methods`